### DISCUSSION TITLE
New Initiative: GitOps Blueprints, call for volunteers

### DISCUSSION THREAD

**Discussion by todaywasawesome** [open-gitops/project#38] [Ideas]
As we've discussed in many sessions, we'd love to have something along the lines of best practices, blueprints, patterns, etc where members of the community and vendors can submit implementation patterns for GitOps using specific tools. This should have a section on the website where people can search and find blueprints based on the tools in use. For example, if a user would like to implement GitOps using AWS, Argo CD, and Vault, it would be awesome if they could find that on the GitOps website. This would also provide a great place for people to contribute what's working well for them. 

This would also provide a good bridge from the principles to implementation and I would think vendors would be excited to demonstrate how their tools can fit into a GitOps flow. 

I'm hoping someone is willing to volunteer to take the lead on this issue and bring a few folks together to make it awesome!

**Comment by lloydchang** [comment]
To future volunteers who answer @todaywasawesome's call to action for [New Initiative: GitOps Blueprints, call for volunteers #38](https://github.com/open-gitops/project/discussions/38):

For your consideration, the following is a suggestion about outreach efforts:

* You may want to search on the Internet for articles with sufficient level of detail about GitOps implementation patterns...

* ... then try outreach efforts by asking articles' authors if the Open GitOps website at https://opengitops.dev/ can reference their articles in the context of GitOps Blueprints

* For example: [Decentralized GitOps over multiple environments at SAP Artificial Intelligence](https://blogs.sap.com/2021/05/06/decentralized-gitops-over-environments/) appears to be at sufficient level detail about implementation patterns:

    * ... explaining how Gardener, Kubernetes, Argo CD, Argo Workflows, Sealed Secrets, AWS, etc. are used in [that context](https://blogs.sap.com/2021/05/06/decentralized-gitops-over-environments/)

* A plausible next step is for a volunteer to comment on [that SAP web site](https://blogs.sap.com/2021/05/06/decentralized-gitops-over-environments/), then ask its author, [Jan Fokken at SAP](https://people.sap.com/jan.fokken), if Open GitOps website at https://opengitops.dev/ can reference the article

Thank you.

**Comment by moshloop** [comment]
I suggest we start with gathering requests for content and individual discussions which can then be voted on for prioritization and to coordinate so that people work on different content pieces

**Comment by tsahiduek** [comment]
Hi,
Is this still relevant?
My colleague (https://github.com/ivallhon) and  I published 2 samples that demonstrate full lifecycle of provisioning an Amazon EKS cluster, bootstrapping it against gitops repository using Flux V2, and will be happy to get feedback, and contribute it here.

The cluster provisioning process, which was designed as part of broader approach for environment management using IaC (in this case it's CDK), that will hold configurations for all environment resources, and not just the EKS cluster.
The sample is available at https://github.com/aws-samples/eks-multi-environment-cdk-pipeline

The GitOps configuration parts demonstrate multiple environment configurations that spans over 3 layers (Infrastructure, configuration, and application). This repository demonstrate monorepo approach for managing both add-on, and applications deployment configuration.
This sample available at https://github.com/aws-samples/flux-eks-gitops-config/
